lib/classes/core_extensions.rb

```diff
@@ -0,0 +1,17 @@
+class Array
+	def all_empty?
+		self.all? { |item| item.to_s.empty? }
+	end 
+
+	def all_same?
+		self.all? { |item| item == self[0] }
+	end
+
+	def any_empty?
+		self.any? { |item| item.to_s.empty? }
+	end
+
+	def none_empty?
+		!any_empty?
+	end
+end
```

Comment:
This is a clever idea but I'm not crazy about the extension of core elements. This discussion enters the realm of personal preference/philosophy. [This is the best info I could find about it ](https://stackoverflow.com/questions/16504369/is-it-proper-ruby-style-to-extend-built-in-classes)

Reply:
Hi, we like it. It also allows us to implement some of the skills we learned in the previous project.

But we really appreciate your comment and perhaps we can discuss at a later date this issue.

Reply:
> This is a clever idea but I'm not crazy about the extension of core elements. This discussion enters the realm of personal preference/philosophy. [This is the best info I could find about it ](https://stackoverflow.com/questions/16504369/is-it-proper-ruby-style-to-extend-built-in-classes)

Thanks for the informative link. I echo Ben Miller's comment:
"[...] Monkey patching would be redefining the functionality of an existing method on the base class with out creating a new type. Basically something like making Array.each() do something different. What you are doing here is not polymorphism, it is inheritance and is the back bone of OOP."